### ISSUE TITLE
(1.13) Teleporting to a home/spawn that is in water from another world puts you in the wrong place.

### ISSUE THREAD

**Issue by Aerodog** [EssentialsX/Essentials#2121]
Within the same world, you get put in the water at the right coordinates. 

From another world, you get put in the nearest air block above the X, Z coordinate.

Re-sending the command will put you in the proper place.

**Comment by mdcfe** [comment]
Apologies for the delay in looking into this, but could you clarify what teleport safety settings you're using so I can try and replicate this?

**Comment by mdcfe** [comment]
I can reproduce this on the following versions:
```java
[16:49:22 INFO]: CONSOLE issued server command: /ess version
[16:49:22 INFO]: Server version: 1.13.2-R0.1-SNAPSHOT git-Paper-487 (MC: 1.13.2)
[16:49:22 INFO]: EssentialsX version: 2.16-pre1.14
[16:49:22 INFO]: LuckPerms version: 4.3.58
[16:49:22 INFO]: Vault version: 1.7.1-b91
[16:49:22 INFO]: EssentialsXProtect version: 2.16-pre1.14
[16:49:22 INFO]: EssentialsXSpawn version: 2.16-pre1.14
[16:49:22 INFO]: EssentialsXChat version: 2.16-pre1.14
[16:49:22 INFO]: EssentialsXGeoIP version: 2.16-pre1.14
[16:49:22 INFO]: EssentialsXAntiBuild version: 2.16-pre1.14
[16:49:22 INFO]: EssentialsXXMPP version: 2.16-pre1.14
```

and the following teleport-safety settings in `config.yml`:
```yaml
# If the teleport destination is unsafe, should players be teleported to the nearest safe location?
# If this is set to true, Essentials will attempt to teleport players close to the intended destination.
# If this is set to false, attempted teleports to unsafe locations will be cancelled with a warning.
teleport-safety: true

# This forcefully disables teleport safety checks without a warning if attempting to teleport to unsafe locations.
# teleport-safety and this option need to be set to true to force teleportation to dangerous locations.
force-disable-teleport-safety: false
```

**Comment by mdcfe** [comment]
This has now been fixed by #2520/https://github.com/EssentialsX/Essentials/commit/74d96ce6249e0c53d25ec7e03197e999d4dcc938 - you can download the latest version from the [build server](https://ci.ender.zone/job/EssentialsX).

You can now use the `is-water-safe: true/false` option in `config.yml` to control whether water is considered "safe" (can be teleported into) or unsafe (EssentialsX will find a safe location).
